lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -160,68 +160,59 @@
+# The following example demonstrates the use of [FinetuningScheduler](https://finetuning-scheduler.readthedocs.io/en/stable/api/finetuning_scheduler.fts.html#finetuning_scheduler.fts.FinetuningScheduler) to fine-tune a small foundational model on the [RTE](https://huggingface.co/datasets/viewer/?dataset=super_glue&config=rte) task of [SuperGLUE](https://super.gluebenchmark.com/). Iterative early-stopping will be applied according to a user-specified schedule.
 #
 
 # %%
 import os
 import warnings
 from datetime import datetime
-from importlib import import_module
 from typing import Any, Dict, List, Optional
 
-import datasets
+from packaging.version import Version
 
 import sentencepiece as sp  # noqa: F401 # isort: split
+import datasets
 import pytorch_lightning as pl
 import torch
+from datasets import logging as datasets_logging
 from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
 from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
 from pytorch_lightning.utilities import rank_zero_warn
-from pytorch_lightning.utilities.cli import _Registry
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from torch.optim.adamw import AdamW
 from torch.optim.lr_scheduler import CosineAnnealingWarmRestarts
 from torch.utils.data import DataLoader
 from transformers import AutoConfig, AutoModelForSequenceClassification, AutoTokenizer
 from transformers import logging as transformers_logging
 from transformers.tokenization_utils_base import BatchEncoding
 
-# %%
-# a couple helper functions to prepare code to work with a user module registry
-MOCK_REGISTRY = _Registry()
-
-
-def mock_register_module(key: str, require_fqn: bool = False) -> List:
-    if key.lower() == "finetuningscheduler":
-        mod = import_module("finetuning_scheduler")
-        MOCK_REGISTRY.register_classes(mod, pl.callbacks.Callback)
-    else:
-        raise MisconfigurationException(f"user module key '{key}' not found")
-    registered_list = []
-    # make registered class available by unqualified class name by default
-    if not require_fqn:
-        for n, c in MOCK_REGISTRY.items():
-            globals()[f"{n}"] = c
-        registered_list = ", ".join([n for n in MOCK_REGISTRY.names])
-    else:
-        registered_list = ", ".join([c.__module__ + "." + c.__name__ for c in MOCK_REGISTRY.classes])
-    print(f"Imported and registered the following callbacks: {registered_list}")
-
+if Version(torch.__version__) == Version("1.12.0") or torch.__version__.startswith("1.12.0"):
```

Comment:
what's up with `startswith(1.12.0)`? like this version will be that?

Reply:
I actually didn't recognize the need for this condition set extension until I installed ``torch`` using this method below for some other testing/dev purposes:
```bash
pip install torch torchvision torchtext --extra-index-url https://download.pytorch.org/whl/cu116;
python -c "import torch;from packaging.version import Version; print(f\"torch version installed: {torch.__version__}\"); print(f\"Version test: {Version(torch.__version__) == Version('1.12.1')}\"); print(f\"startswith test: {torch.__version__.startswith('1.12.1')}\");"
torch version installed: 1.12.1+cu116
Version test: False
startswith test: True
```
Unfortunate ``Version`` doesn't recognize that version style but I get it, interesting anyway, every day has its own tutorial :smile: ! I included both test conditions just in case.